### PR TITLE
359 improve plotting

### PR DESCRIPTION
**Pull request by menriquez-IDM**
This PR significantly improves the `tbsim.plots` module by modernizing the plotting API, enhancing usability, and adding comprehensive test coverage. The main changes include:

- **Flexible input handling**: Supports MultiSim, single Sim, or dict of flat results
- **Improved metric selection**: Pandas-style filtering with `items`, `like`, `regex`, and `exclude` options
- **Better error handling**: Graceful handling of missing metrics, invalid inputs, and edge cases
- **Updated documentation**: All examples, tutorials, and notebooks updated to use the new API
- **Renamed function**: `plot_results()` → `plot()` for a cleaner, more intuitive API
- **Enhanced metric labels**: Plots now display human-readable labels (e.g., "TB: New infections") from Result objects instead of technical metric names (e.g., "tb_new_infections")

## Changes by File
PLOTS FILE: https://github.com/starsimhub/tbsim/blob/359-improve-plotting/tbsim/plots.py
### Core Module
- **`tbsim/plots.py`**: Complete refactor of plotting utilities
  - Renamed `plot_results()` → `plot()`
  - Added automatic extraction of Result object labels via `full_label` or `label` attributes
  - Improved parameter naming (`output_dir` instead of `outdir`, `theme` instead of `dark`)
  - Enhanced metric selection with pandas-style filtering
  - Better error handling and validation
  - Fixed colormap cycling bug

### Tests
- **`tests/test_plots.py`**: Enhanced test coverage
  - Added tests for MultiSim, single Sim, and flat results
  - Tests for metric selection, validation, and edge cases
  - Tests for helper functions (`_as_1d_xy`, `_fill_missing_metrics`, etc.)

### Examples
Updated all example scripts to use `tbsim.plot()`:
- `tbsim_examples/run_health_seeking.py`
- `tbsim_examples/run_malnutrition.py`
- `tbsim_examples/run_tb.py`
- `tbsim_examples/run_tb_interventions.py`
- `tbsim_examples/run_tb_lshtm.py`
- `tbsim_examples/run_tbhiv.py`

### Documentation
Updated tutorials and documentation:
- `docs/tutorials/basic_tbrun.md`
- `docs/tutorials/malnutrition_tutorial.ipynb`
- `docs/tutorials/tb_interventions_tutorial.ipynb`
- `docs/tutorials/tuberculosis_sim.ipynb`

### 1. Human-Readable Plot Labels
Plots now automatically extract and display the `full_label` property from Result objects (e.g., "TB: Prevalence" instead of "tb_prevalence"), making visualizations more professional and user-friendly.

### 2. Simplified API
```python
# Before
tbsim.plot_results(results, dark=True, savefig=True, outdir='results')

# After
sim.tb_lshtm_acute.plot()   # For a solo sim
or
tbsim.plot(msim, theme='dark', savefig=True, output_dir='results') # For multisims
```

### 3. Flexible Metric Selection
```python
# Select all metrics
tbsim.plot(msim)

# Select specific metrics
tbsim.plot(msim, select=['incidence', 'prevalence'])

# Select by pattern
tbsim.plot(msim, select=dict(like='prev'))

# Select by regex
tbsim.plot(msim, select=dict(regex=r'^tb_.*'))
```

### 4. Better Error Handling
- Validates input types (MultiSim, Sim, or dict)
- Handles missing metrics gracefully (fills with zeros)
- Provides informative error messages for invalid inputs

## Testing

All existing tests pass, and new tests provide comprehensive coverage:
- MultiSim plotting
- Single Sim plotting
- Flat results dict plotting
- Metric selection and filtering
- Edge cases and error handling

## Migration Guide

For existing code using `plot_results()`:

```python
# Old
tbsim.plot_results(results, dark=True, savefig=True, outdir='results')

# New
tbsim.plot(results, theme='dark', savefig=True, output_dir='results')
```

---
scenarios_20260311_153021

---
scenarios_20260311_153003

---
scenarios_20260311_153029

---
scenarios_20260311_153033

### CODE REVIEW

File: tbsim/plots.py [hunk 1]
```diff
@@ -1,458 +1,422 @@
...
+    select=None,
+    title='',
+    savefig=False,
+    output_dir=None,
+    theme='light',
+    n_cols=6,
+    row_height=1.5,
+    style=None,
+    show=True,
+):
```
Comment (cliffckerr) [inline tbsim/plots.py]: Starsim doesn't use this style of doing arguments; please fill the line width as before.

File: tbsim/plots.py [hunk 2]
```diff
@@ -1,458 +1,422 @@
...
+    return out
 
-    """
 
+def _parse_exclude_patterns(items):
+    """Return list of patterns from '~Something' or list like ['~None','~temp'] (strip leading ~)."""
+    if not items:
+        return []
+    it = (items,) if isinstance(items, str) else items
+    return [p[1:] if isinstance(p, str) and p.startswith('~') else str(p) for p in it if p]
```
Comment (cliffckerr) [inline tbsim/plots.py]: This is cool! We could incorporate it in Starsim's plotting logic

File: tbsim/plots.py [hunk 3]
```diff
@@ -1,458 +1,422 @@
...
-    n_rows = int(np.ceil(len(metrics) / n_cols))
-    fig, axs = plt.subplots(n_rows, n_cols, figsize=(4.5*n_cols, heightfold*n_rows+1))
-    axs = np.array(axs).flatten()
+def _as_list(values):
+    """Return a list from scalar/list-like input."""
+    if values is None:
+        return []
+    if isinstance(values, (list, tuple, set)):
+        return list(values)
+    return [values]
```
Comment (cliffckerr) [inline tbsim/plots.py]: Use `sc.tolist()` as a drop-in replacement instead

File: tests/test_plots.py [hunk 4]
```diff
@@ -1,157 +1,133 @@
...
- 
- 
\ No newline at end of file
+
+
+def make_msim(use_alt_model=False):
+    pars = dict(dt=ss.days(14), start=ss.date('2000-01-01'), stop=ss.date('2002-12-31'), verbose=0)
+    sims = []
+    for i, label in enumerate(('Scenario A', 'Scenario B', 'Scenario C')):
+        disease_cls = ss.SIS if (use_alt_model and i == 0) else ss.SIR
```
Comment (cliffckerr) [inline tests/test_plots.py]: Why is this using something other than TB as the disease?

File: tests/test_plots.py [hunk 5]
```diff
@@ -1,157 +1,133 @@
...
+
+
+@pytest.fixture(scope='module')
+def mixed_msim():
+    return make_msim(use_alt_model=True)
+
+
+@pytest.fixture(scope='module')
+def flat_results(msim):
+    return make_flat_results(msim)
```
Comment (cliffckerr) [inline tests/test_plots.py]: This is all very complicated ... would be nice to keep the tests simple and flat without pytest fixtures.

**Review by cliffckerr — CHANGES_REQUESTED** [review]
Thanks for this PR, @menriquez-IDM . There's a lot of good changes in here, however I have two major comments that would be worth addressing before merging:

## Inherit from `tbsim.Sim`
In `rc0.7`, there's a new `tbsim.Sim` class which inherits from `starsim.Sim`. I feel like a lot of the plotting logic here is duplicating what's already in `sim.plot()` (and some of it's duplicating what's in `MultiSim.plot()`. I would like to see this code inherit from that, like this:
```py
def plot(self, <various args):
  <do any custom logic you need>
  return super().plot(<the custom args you need>)
```
If `sim.plot()` isn't flexible enough to handle what you need, then we can update it!

## Reduce complexity
There's a lot of code and complexity in here... a lot of is doing some cool things, but my sense is that it creates more technical debt than we'll want to deal with for a plotting routine. Hopefully inheriting from `Sim.plot` will simplify things a lot, but let's try to keep things minimal beyond that.

File: tests/test_plots.py (hunk 5 above)
Comment (menriquez-IDM) [reply tests/test_plots.py]: man, this is very minimal, I am really downsizing here.
